File: oldsrc/discovery-core/src/main/java/org/endeavourhealth/discovery/core/definition/models/ItemType.java
package org.endeavourhealth.discovery.core.definition.models;

import java.util.HashMap;
import java.util.Map;

public enum ItemType {

    FOLDER(0),
    REPORT(1),
    QUERY(2),
    TEST(3),
    DATASOURCE(4),
    CODESET(5);

    private int value;

    private ItemType(int value){
        this.value = value;
    }

    private static final Map<Integer, ItemType> lookup = new HashMap<>();

    static {
        for (ItemType d : ItemType.values())
            lookup.put(d.getValue(), d);
    }

    public static ItemType get(int value) {
        return lookup.get(value);
    }

    public int getValue() {
        return value;
    }
};

File: src/enterprise-web/src/main/web/app/models/ItemType.ts
/// <reference path="../../typings/tsd.d.ts" />

module app.models {
	export enum ItemType {
		Folder,			// 0
		Report,			// 1
		Query,			// 2
		Test,				// 3
		Datasource,	// 4
		CodeSet,		// 5
		ListOutput	// 6
	}

	export function itemTypeIdToString() {
		return function (input:number):string {
			switch (input) {
				case ItemType.Folder:
					return 'Folder';
				case ItemType.Report:
					return 'Report';
				case ItemType.Query:
					return 'Query';
				case ItemType.Test:
					return 'Test';
				case ItemType.Datasource:
					return 'Datasource';
				case ItemType.CodeSet:
					return 'Code set';
				case ItemType.ListOutput:
					return 'List report';
			}
		};
	}

	export function itemTypeIdToIcon() {
		return function (input:number):string {
			switch (input) {
				case ItemType.Folder:
					return 'fa-folder-open';
				case ItemType.Report:
					return 'fa-file';
				case ItemType.Query:
					return 'fa-question-circle';
				case ItemType.Test:
					return 'fa-random';
				case ItemType.Datasource:
					return 'fa-database';
				case ItemType.CodeSet:
					return 'fa-tags';
				case ItemType.ListOutput:
					return 'fa-list-alt';
			}
		};
	}

	angular
		.module('app.models')
		.filter('itemTypeIdToString', itemTypeIdToString)
		.filter('itemTypeIdToIcon', itemTypeIdToIcon);
}

